src/core/server/saved_objects/export/collect_exported_objects.ts

```diff
@@ -126,3 +165,68 @@ const fetchReferences = async ({
       .map((obj) => ({ type: obj.type, id: obj.id })),
   };
 };
+
+const buildTransforms = (typeRegistry: ISavedObjectTypeRegistry) =>
+  typeRegistry.getAllTypes().reduce((transforms, type) => {
+    if (type.management?.onExport) {
+      return {
+        ...transforms,
+        [type.name]: type.management.onExport,
+      };
+    }
+    return transforms;
+  }, {} as Record<string, SavedObjectsExportTransform>);
+
+const buildIsExportable = (
+  typeRegistry: ISavedObjectTypeRegistry
+): SavedObjectsExportablePredicate<any> => {
+  const exportablePerType = typeRegistry.getAllTypes().reduce((transforms, type) => {
+    if (type.management?.isExportable) {
+      transforms[type.name] = type.management.isExportable;
+    }
+    return transforms;
+  }, {} as Record<string, SavedObjectsExportablePredicate>);
+
+  return (obj: SavedObject) => {
+    const typePredicate = exportablePerType[obj.type];
+    return typePredicate ? typePredicate(obj) : true;
+  };
+};
+
+const splitByExportability = async (
+  objects: SavedObject[],
+  isExportable: SavedObjectsExportablePredicate<any>
+) => {
+  const exportableObjects: SavedObject[] = [];
+  const nonExportableObjects: SavedObject[] = [];
+  await Promise.all(
```

Comment:
I'm not too familiar with how the batching works here, but is there any concern about possibly initiating an ES or other external request(s) for each of these objects all at once? I believe in the ES case, the client's pooling & queueing mechanisms should handle this for us, but in the case of an external request, we'd be relying on the plugin to be smart here. Not sure that's an assumption we can easily make.

But again, if the batching done at the calling code is reasonable enough, this may not be an issue.

Reply:
> Do we have a known use case for async isExportable implementations? If not, this could be something we opt not to support at this time to avoid any performance issues with this new hook.

Not really, atm the only identified use case is based on the object's properties, so for now I will just make the hook synchronous.